ast/src/analyzed/mod.rs

```diff
@@ -310,6 +310,10 @@ impl<T> Identity<T> {
         assert_eq!(self.kind, IdentityKind::Polynomial);
         self.left.selector.as_ref().unwrap()
     }
+
+    pub fn contains_next_ref(&self) -> bool {
```

Comment:
Actually: is there a way to use `previsit_expressions_in_identity_mut` defined in `utils`? Then we don't need to define this for `SelectedExpressions`